src/core/listener.c

```diff
@@ -643,6 +644,57 @@ QuicListenerClaimConnection(
         return FALSE;
     }
 
+    const uint8_t* NewNegotiatedAlpn = Event.NEW_CONNECTION.NewNegotiatedAlpn;
+    if (NewNegotiatedAlpn) {
+        const uint8_t* ListStart = Info->ClientAlpnList;
+        const uint8_t* ListEnd = Info->ClientAlpnList + Info->ClientAlpnListLength;
+        if (!(ListStart <= NewNegotiatedAlpn &&
+            ListEnd > NewNegotiatedAlpn &&
+            NewNegotiatedAlpn + NewNegotiatedAlpn[0] + sizeof(uint8_t) <= ListEnd)
```

Comment:
I find the ~(x && y && ...) format hard to parse. Would you mind inverting it?

Reply:
I'll invert it